### PR TITLE
refactor: Replace deprecated `substr` with `substring`

### LINKED ISSUE #8645: Server uses deprecated substr()
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the [latest version of Parse Server](https://github.com/parse-community/parse-server/releases). 

### Issue Description

`substr()` has been deprecated by `substring()`, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substr. 

### Steps to reproduce

The deprecation shows when relevant files are opened in IDE's like VS Code

### Actual Outcome

Deprecation warning

### Expected Outcome

No deprecation warning

### Environment

Server
- Parse Server version: `6.2.0`
- Operating system: `Linux`
- Local or remote host (AWS, Azure, Google Cloud, Heroku, Digital Ocean, etc): `Local`

Database
- System (MongoDB or Postgres): `Postgres`
- Database version: `15`
- Local or remote host (MongoDB Atlas, mLab, AWS, Azure, Google Cloud, etc): `Local`

Client
- SDK (iOS, Android, JavaScript, PHP, Unity, etc): `N/A`
- SDK version: `N/A`

### PR DESCRIPTION
**Pull request by cbaker6**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/parse-server/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/parse-server/blob/alpha/LICENSE).
- Link this pull request to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

## Issue

`substr()` has been deprecated.

Closes: #8645

## Approach

- [x] Use `substring()` instead and fix issues where needed
- [x] Refactor `rotateEncryptionKey()` in `GridFSBucketAdapter.js` to use async/await since already refactoring `substr()` in the same file 

## Tasks

- [x] Ensure tests pass

### CODE REVIEW

**Comment by cbaker6** [comment]
@mtrezza @dblythy this is ready for review

File: src/Adapters/Storage/Postgres/PostgresStorageAdapter.js [hunk 1]
```diff
@@ -1921,14 +1921,14 @@ export class PostgresStorageAdapter implements StorageAdapter {
         };
       }
       if (object[fieldName] && schema.fields[fieldName].type === 'Polygon') {
-        let coords = object[fieldName];
-        coords = coords.substr(2, coords.length - 4).split('),(');
-        coords = coords.map(point => {
+        let coords = new String(object[fieldName]);
+        coords = coords.substring(2, coords.length - 2).split('),(');
+        const updatedCoords = coords.map(point => {
```
Comment (mtrezza) [inline src/Adapters/Storage/Postgres/PostgresStorageAdapter.js:1926]: This change seem to be more than replacing `substr`. Why does the subtraction change from 4 to 2?

**Review by mtrezza — APPROVED** [review]
Looks good

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.3.0-alpha.2](https://github.com/parse-community/parse-server/releases/tag/6.3.0-alpha.2)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.4.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/6.4.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.4.0-alpha.1](https://github.com/parse-community/parse-server/releases/tag/6.4.0-alpha.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.4.0](https://github.com/parse-community/parse-server/releases/tag/6.4.0)

**Pull request merged**